node/http/src/types.rs

```diff
@@ -304,8 +303,19 @@ pub struct GetTransactionRequest {
 }
 
 #[derive(Serialize, Deserialize)]
-pub struct TransactionStatusResponse {
-    pub status: TransactionStatus,
+pub enum FinalTransactionStatus {
```

Comment:
this status should live with transaction, as it is not relative to this particular interface

Reply:
well, for me - this is only an RPC stuff - I would ideally push this to frontend to query instead of computing here.
E.g. we don't need this logic anywhere else.

Reply:
maybe this is a premature optimization but i feel weird binding this kind of stuff with a specific network interface. one example where i think this can be used is when we are pushing similar data that will be fetched from this http layer via websockets.. maybe this means we need another layer on top of client to support views?

Reply:
Moved it into shard chain for now.